### PR TITLE
Support ResourceGroupNameEquals and ResourceNameEquals as parameters in Find-AzureRmResource

### PR DESCRIPTION
**Pull request by TianoMS**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/dev/documentation/platyPSHelp-documentation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/dev/documentation/breaking-changes.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://gist.github.com/markcowl/338e16fe5c8bbf195aff9f8af0db585d#what-is-the-change).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThrough` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/dev/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.
…in Find-AzureRmResource.

### CODE REVIEW

**Comment by TianoMS** [comment]
On demand job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1327/

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/QueryFilterBuilder.cs [hunk 1]
```diff
@@ -27,15 +27,15 @@ public static class QueryFilterBuilder
...
-        /// <param name="subscriptionIds">The subscriptions to query.</param>
+        /// <param name="subscriptionId">The subscription to query.</param>
         /// <param name="resourceGroup">The name of the resource group/</param>
         /// <param name="resourceType">The resource type.</param>
         /// <param name="resourceName">The resource name.</param>
         /// <param name="tagName">The tag name.</param>
         /// <param name="tagValue">The tag value.</param>
         /// <param name="filter">The filter.</param>
         public static string CreateFilter(
-            Guid[] subscriptionIds,
```
Comment (vivsriaus) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/QueryFilterBuilder.cs]: Why are we removing the support for multiple subIds?

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Resource/FindAzureResourceCmdlet.cs [hunk 2]
```diff
@@ -469,11 +492,19 @@ private bool IsSubscriptionLevelQuery()
...
                 this.ResourceType != null ||
                 this.ExtensionResourceType != null);
         }
+
+        /// <summary>
+        /// Returns true if resource group name is availabe in parameters.
+        /// </summary>
+        private bool ResourceGroupNameAvailable()
+        {
+            return this.ResourceGroupNameContains != null || this.ResourceGroupNameEquals != null;
```
Comment (vivsriaus) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Resource/FindAzureResourceCmdlet.cs]: string.IsNullOrEmpty is better here, no?

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/QueryFilterBuilder.cs (hunk 1 above)
Comment (TianoMS) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Components/QueryFilterBuilder.cs]: we don't have any scenario that has mutiple subsIds in our repo now.

**Comment by cormacpayne** [comment]
@TianoMS please update the [ChangeLog](https://github.com/Azure/azure-powershell/blob/dev/src/ResourceManager/Resources/ChangeLog.md) with changes you have made in this PR under the section **Current Release**

**Comment by cormacpayne** [comment]
On demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1332/

**Comment by vivsriaus** [comment]
LGTM thanks! @markcowl @cormacpayne

**Comment by cormacpayne** [comment]
LGTM

**Pull request merged**